lib/swoosh/email.ex

```diff
@@ -456,12 +458,46 @@ defmodule Swoosh.Email do
   ## Examples
 
       iex> new() |> assign(:username, "ironman")
-      %Swoosh.Email{assigns: %{username: "ironman"}, attachments: nil, bcc: [],
+      %Swoosh.Email{assigns: %{username: "ironman"}, attachments: [], bcc: [],
        cc: [], from: nil, headers: %{}, html_body: nil, private: %{},
        provider_options: %{}, reply_to: nil, subject: "", text_body: nil, to: []}
   """
   @spec assign(t, atom, any) :: t
   def assign(%__MODULE__{assigns: assigns} = email, key, value) when is_atom(key) do
     %{email | assigns: Map.put(assigns, key, value)}
   end
+
+  @doc ~S"""
+  Add a new attachment in the email.
+
+  ## Examples
+
+      iex> new() |> attachment("/data/att.zip")
+      %Swoosh.Email{assigns: %{}, bcc: [], cc: [], from: nil,
+       headers: %{}, html_body: nil, private: %{}, provider_options: %{},
+       reply_to: nil, subject: "", text_body: nil, to: [],
+       attachments: [%Swoosh.Attachment{path: "/data/att.zip", content_type: "application/zip", filename: "att.zip"}]}
+      iex> new() |> attachment(Swoosh.Attachment.new("/data/att.zip"))
+      %Swoosh.Email{assigns: %{}, bcc: [], cc: [], from: nil,
+       headers: %{}, html_body: nil, private: %{}, provider_options: %{},
+       reply_to: nil, subject: "", text_body: nil, to: [],
+       attachments: [%Swoosh.Attachment{path: "/data/att.zip", content_type: "application/zip", filename: "att.zip"}]}
+      iex> new() |> attachment(%Plug.Upload{path: "/data/abcdefg", content_type: "test/type", filename: "att.zip"})
+      %Swoosh.Email{assigns: %{}, bcc: [], cc: [], from: nil,
+       headers: %{}, html_body: nil, private: %{}, provider_options: %{},
+       reply_to: nil, subject: "", text_body: nil, to: [],
+       attachments: [%Swoosh.Attachment{path: "/data/abcdefg", content_type: "test/type", filename: "att.zip"}]}
+  """
+  @spec attachment(t, binary | Swoosh.Attachment.t) :: t
+  def attachment(%__MODULE__{attachments: attachments} = email, <<path::binary>>) do
```

Comment:
@princemaple and I discussed this on Slack I'm ok with keeping `attachment` to be more consistent with the rest of the API.